ligo/defaults.mligo

```diff
@@ -0,0 +1,67 @@
+// SPDX-FileCopyrightText: 2021 Arthur Breitman
+// SPDX-License-Identifier: LicenseRef-MIT-Arthur-Breitman
+
+#include "types.mligo"
+#include "consts.mligo"
+
+// ----------------------------------------------------------------------------
+// Pow function, implemented from haskell (^)
+// ----------------------------------------------------------------------------
+
+let quot (a, b: int * int): int =
+  match ediv a b with
+      Some (quotient, _rem) -> quotient
+    | None -> (failwith("`b` can't be 0."): int)
+
+let rec pow_g (x, y, z: int * int * int): int =
+      if (y mod 2 = 0n) then pow_g (x * x, quot (y, 2), z)
+      else if (y = 1) then x * z
+      else pow_g (x * x, quot (y, 2), x * z)
+
+let rec pow_f (x, y: int * int): int =
+      if (y mod 2 = 0n) then pow_f (x * x, y / 2)
+      else if (y = 1) then x
+      else pow_g ( x * x, quot (y, 2), x)
+
+let pow (x, y: int * int) : int =
+  if (y < 0) then (failwith("Negative exponent"): int)
+  else if (y = 0) then 1
+  else pow_f (x, y)
+
+// ----------------------------------------------------------------------------
+
+
+let default_storage : storage =
+  let infinity : int = pow (10, 100) in
+
+  // Note: Taken form ligo/tests/tests.mligo
+  let infinity_tick =
+    { prev = { i = -infinity }
+    ; next = { i = infinity }
+    ; liquidity_net = 0
+    ; seconds_outside = 0n
+    ; seconds_per_liquidity_outside = 0n
+    ; n_positions = 0n
+    ; fee_growth_outside = {x = 0n; y = 0n}
+    ; sqrt_price = 0n
```

Comment:
Actually, just noticed some of these fields have different values from `ligo/tests/tests.mligo`. Namely, `n_positions`, `sqrt_price`

Reply:
Initially, I make it the same as `tests.mligo`, but turn out that it may not be necessary.

I forgot to remove the comment. If you think it is better to use the same value as the `tests.mligo`, I'll make the changes, otherwise I'll remove the comment.

Reply:
I think setting `n_positions` to `1` is necessary, otherwise a user could set a position at `(-inf, +inf)`, then close the position (which would uninitialize the `-inf` and `+inf` ticks), and leave the contract in an irreparable state.

Reply:
I think `sqrt_price` should have the value corresponding to its tick index (using the equation from the spec), if we want positions at `(-inf, +inf)` to work correctly.

Reply:
Since `sqrt(P) = sqrt(X/Y)`, I'm not quite what would the value of X and Y for -inf and +inf. Do we just set a fixed value for X and Y? Hope to get your advice on this.

Reply:
That's the equation for the _spot price_ (the price at a specific point in time, calculated from the reserves of `x` and `y` at that moment).

The price for a tick `i` is given by:

```
p(i) = exp(0.0001)^i
```

Reply:
I see, this is still quite confusing to me. I check `tests.mligo` and seems like the formula used is a bit different: `Round[Exp[5/100000*(2^20-1)]*2^80]`. 

I try your formula quickly in a javascript console and got this:
```
Math.exp(0.0001) ^ (10 ^ 100)
111
```
Which seems a bit strange, maybe I get the parenthesis wrong, I try this instead and got
```
Math.exp((0.0001)^ (10^100))
5.92097202766467e+47
```
Since I'm quite lack in this, hope to get your advice on which one is correct.

Reply:
> I  see, this is still quite confusing to me. I check tests.mligo and seems like the formula used is a bit different: `Round[Exp[5/100000*(2^20-1)]*2^80]`.

I ran some calculations, and that equation seems to be equivalent to `p(i) = exp(0.0001)^i` (_taking into account we store the price as a square root * 2^80_).

For example, for the tick `const_max_tick = 1048575`, we should get:

```
sqrt_price = sqrt(exp(0.0001)^1048575) * 2^80
           = 71107673757466966990985105047137336834554167630.0560224...
           ~ 71107673757466966990985105047137336834554167630
```

And for `-const_max_tick = -1048575`, we should get:

```
sqrt_price = sqrt(exp(0.0001)^-1048575) * 2^80
           = 20.5533...
           ~ 21
```

Which matches the values in `ligo/tests/tests.mligo`.

(You can use a calculator [like this one](https://www.mathsisfun.com/calculator-precision.html) to avoid results in scientific notation)

Reply:
I see, I tried the calculator with `sqrt(exp(0.0001)^(10^100)) * 2^80`. It results in literally infinity. 

Should we use some other value (or just use `1048575` like the test)?

Reply:
Hmm good question, let's resume this conversation in Slack and get other people's opinions on this!

Reply:
@pasqu4le Maybe you missed this, what do you think about this?

Reply:
@dcastro I've updated this now, you can take a look again.